src/core/lib/event_engine/default_event_engine.cc

```diff
@@ -57,13 +61,21 @@ void EventEngineFactoryReset() {
   g_event_engine->reset();
 }
 
-std::unique_ptr<EventEngine> CreateEventEngine() {
+std::unique_ptr<EventEngine> CreateEventEngineInner() {
```

Comment:
I'm not sure the inner method is helping anything here. Alternative:

```
std::unique_ptr<EventEngine> CreateEventEngine() {
  if (auto* factory = g_event_engine_factory.load()) {
    return (*factory)();
  }
#ifdef GRPC_MAXIMIZE_THREADYNESS
  return std::make_unique<ThreadyEventEngine>(DefaultEventEngineFactory());
#endif
  return DefaultEventEngineFactory();
}
```

Reply:
That skips the wrapping in the case of a custom event engine (which I think is undesirable)

Reply:
Yep, my mistake, I misread the code.